src/components/addSourceWizard/hardcodedComponents/azure/costManagement.js

```diff
@@ -124,19 +125,78 @@ const InternalReaderRoleDescription = () => {
 
   let scope = application?.extra?.scope || `/subscriptions/${application?.extra?.subscription_id}`;
 
-  return (
-    <TextContent>
-      <Text component={TextVariants.p}>
-        {intl.formatMessage({
-          id: 'cost.azure.createReaderRole',
-          defaultMessage: 'Run the following command in Cloud Shell to create a Cost Management Reader role:',
-        })}
-      </Text>
-      <ClipboardCopy>
-        {`az role assignment create --assignee "${authentication?.username}" --role "Cost Management Reader" --scope "${scope}"`}
-      </ClipboardCopy>
-    </TextContent>
-  );
+  if (scope.includes('billingAccounts')) {
+    if (scope.includes('enrollmentAccounts')) {
+      return (
+        <TextContent>
+          <Text component={TextVariants.p}>
+            {intl.formatMessage(
+              {
+                id: 'cost.azure.setupEAReaderRole',
+                defaultMessage:
+                  'Launch the Azure Enterprise Portal and give the service principal created above Administrator role on the associated account.  {link}',
+              },
+              {
+                link: (
+                  <Text key="link" rel="noopener noreferrer" target="_blank" component={TextVariants.a} href={AZURE_ROLES_URL}>
+                    {intl.formatMessage({
+                      id: 'wizard.learnMore defaultMessage=Learn more',
+                      defaultMessage: 'Learn more',
+                    })}
+                  </Text>
+                ),
+              }
+            )}
+          </Text>
+        </TextContent>
+      );
+    } else {
+      let mcaRole = 'Billing account reader';
+      if (scope.includes('invoiceSections')) {
+        mcaRole = 'Invoice section reader';
+      } else if (scope.includes('billingProfiles')) {
+        mcaRole = 'Billing profile reader';
+      }
```

Comment:
This can be simplified as
```suggestion
      const mcaRole = scope.includes('invoiceSections') ? 'Invoice section reader' : `Billing ${scope.includes('billingProfiles') ? 'profile' : 'account'} reader`;
```